### PR TITLE
Notice for inheritance

### PR DESCRIPTION
**Pull request by wongoo**
**What this PR does**:
Notice for inheritance

**Which issue(s) this PR fixes**:
fix https://github.com/apache/dubbo-go-hessian2/issues/157

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -225,3 +221,34 @@ The encoded bytes of the struct `MyUser` is as following:
  00000030  4e 75 6d 62 65 72 60 08  75 73 65 72 6e 61 6d 65  |Number`.username|
  00000040  0c 30 31 30 2d 31 32 33  34 35 36 37 38           |.010-12345678|
 ```
+
+## Notice for inheritance
+
+`go-hessian2` support inheritance struct, but the following situations should be avoided.
```
Comment (AlexStocks) [inline README.md]: supports

File: README.md [hunk 2]
```diff
@@ -225,3 +221,34 @@ The encoded bytes of the struct `MyUser` is as following:
...
+
+### Avoid fields with the same name in multiple parent struct
```
Comment (AlexStocks) [inline README.md]: maybe using */+/- as prefix maybe better

File: README.md [hunk 3]
```diff
@@ -225,3 +221,34 @@ The encoded bytes of the struct `MyUser` is as following:
...
+```go
+type A struct { Name string }
+type B struct { Name string }
+type C struct {
+	A
+	B
+}
+```
+
+### Avoid inheritance for a pointer of struct
```
Comment (AlexStocks) [inline README.md]: maybe using */+/- as prefix maybe better

**Comment by AlexStocks** [comment]
LGTM

**Comment by fangyincheng** [comment]
is not the description of the anonymous structure field in this pr?

**Comment by wongoo** [comment]
> is not the description of the anonymous structure field in this pr?

it's for https://github.com/apache/dubbo-go-hessian2/issues/157

**Pull request merged**